src/verification/mod.rs

```diff
@@ -58,10 +41,13 @@ fn unsupported_server_name() -> rustls::Error {
     rustls::Error::UnsupportedNameType
 }
 
-// Certificate data error shorthand
+// Unknown certificate error shorthand. Used when we need to construct an "Other" certificate
+// error with a platform specific error message.
 #[cfg(any(windows, target_os = "android", target_os = "macos", target_os = "ios"))]
 fn invalid_certificate(reason: impl Into<String>) -> rustls::Error {
-    rustls::Error::InvalidCertificateData(reason.into())
+    rustls::Error::InvalidCertificate(rustls::CertificateError::Other(std::sync::Arc::from(
+        Box::from(reason.into()),
```

Comment:
This might be reducible to `reason.into()` given [the implementation](https://doc.rust-lang.org/std/error/trait.Error.html#impl-From%3CString%3E-for-Box%3Cdyn%20Error%20+%20%27static%2C%20Global%3E) of `From<String> for Box<Error>`.

Reply:
Hmm, no luck (or I'm holding it wrong):

```rust
    rustls::Error::InvalidCertificate(rustls::CertificateError::Other(std::sync::Arc::from(
        reason.into(),
    )))
```
->
```
error[E0277]: the trait bound `std::sync::Arc<dyn std::error::Error + std::marker::Send + std::marker::Sync>: std::convert::From<std::string::String>` is not satisfied
```

Reply:
It looks like you were right and I was incorrect here. Its a bit unfortunate that `Other` error reporting with `String` now requires double allocation now but I don't think there's anything that can be done about it.